cuda_core/cuda/core/experimental/_program.py

```diff
@@ -433,6 +503,27 @@ def _translate_program_options(self, options: ProgramOptions) -> LinkerOptions:
             ptxas_options=options.ptxas_options,
         )
 
+    def _translate_program_options_to_nvvm(self, options: ProgramOptions) -> list[str]:
+        """Translate ProgramOptions to NVVM-specific compilation options."""
+        nvvm_options = []
+
+        if options.arch is not None:
+            arch = options.arch
+            if arch.startswith("sm_"):
+                arch = f"compute_{arch[3:]}"
+            nvvm_options.append(f"-arch={arch}")
+        else:
+            major, minor = Device().compute_capability
+            nvvm_options.append(f"-arch=compute_{major}{minor}")
```

Comment:
IIRC we already detected current arch in `ProgramOptions` (plz double check), so this is not needed
```suggestion
```

Reply:
I get it now why you need this. Let's change `--gpu-architecture` to `-arch` here
https://github.com/NVIDIA/cuda-python/blob/d64120bb6733391989174456daaf18643ea29711/cuda_core/cuda/core/experimental/_program.py#L231-L236
so that all backends can use this. Then the above deletion can be made.